hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/DistCpSync.java

```diff
@@ -563,10 +589,27 @@ private Path translateRenamedPath(Path sourcePath,
     } else {
       List<DiffInfo> renameDiffsList =
           diffMap.get(SnapshotDiffReport.DiffType.RENAME);
+      List<DiffInfo> deletedDirDiffsList =
+          diffMap.get(SnapshotDiffReport.DiffType.DELETE);
```

Comment:
Will this list hold all files and directories that have been deleted, eg: if I delete a directory with 1000 entries, will this end up with 1001 entries?

Then in `isParentOrSelfMarkedDeleted()` we need to scan this list. Could this list be very large and cause a performance problem when scanning it over and over for each entry in the diffList?

Reply:
The list will hold all the entries that are marked deleted. For single directory with 1000 files when gets deleted, the snapshot diff report itself will just have 1 deleted entry for the directory.

It does scan the whole rename list already. With this, it needs to scan the deleted list as well and hence it can bring performance problem depending how many entries are marked deleted.

I have modified the logic now to build a deleted list for entried which have been marked deleted only bcoz of rename to an excluded path (excluded by filter). This should limit the scans.